File: backend/app/schemas/knowledge_summary.py
from datetime import datetime

from pydantic import BaseModel, Field


class KnowledgeSummaryRead(BaseModel):
    summary: str | None = None
    generated_at: datetime | None = None
    has_summary: bool


class KnowledgeSummaryUpdate(BaseModel):
    summary: str | None = None


class KnowledgeSummaryPromptConfig(BaseModel):
    default_system_prompt: str
    custom_system_prompt: str | None = None
    uses_custom: bool


class KnowledgeSummaryPromptUpdate(BaseModel):
    system_prompt: str | None = None


class KnowledgeSummaryResult(BaseModel):
    summary: str = Field(min_length=1)
    generated_at: datetime

File: frontend/src/types/knowledge_summary.ts
export type KnowledgeSummaryRead = {
  summary: string | null;
  generated_at: string | null;
  has_summary: boolean;
};

export type KnowledgeSummaryUpdate = {
  summary: string | null;
};

export type KnowledgeSummaryResult = {
  summary: string;
  generated_at: string;
};

export type KnowledgeSummaryPromptConfig = {
  default_system_prompt: string;
  custom_system_prompt: string | null;
  uses_custom: boolean;
};

export type KnowledgeSummaryPromptUpdate = {
  system_prompt: string | null;
};

export function formatKnowledgeSummaryDate(iso: string): string {
  return new Date(iso).toLocaleString(undefined, {
    dateStyle: "medium",
    timeStyle: "short",
  });
}
